nibabel/spatialimages.py

```diff
@@ -325,6 +326,23 @@ class SpatialImage(DataobjImage):
     ''' Template class for volumetric (3D/4D) images '''
     header_class = SpatialHeader
 
+    class Slicer(object):
+        def __init__(self, img):
+            self.img = img
+
+        def __getitem__(self, idx):
+            idx = self._validate(idx)
+            dataobj = self.img.dataobj[idx]
+            affine = self.img.slice_affine(idx)
+            return self.img.__class__(dataobj, affine, self.img.header)
+
+        def _validate(self, idx):
+            idx = canonical_slicers(idx, self.img.shape)
+            for slicer in idx:
+                if isinstance(slicer, int):
+                    raise IndexError("Cannot slice image with integer")
```

Comment:
Just pointing this out: I don't want to slice an image like `img.slicer[0, :, :]`, since a numpy user would expect to lose a dimension in that case, and we don't want to drop dimensions. So I'm pushing the `[0]` syntax.

Let me know if y'all disagree.

Reply:
We don't want to drop dimensions because Nifti expects at least three dimensions?  I could well imagine wanting to drop the fourth dimension with `img[..., 1]` for example.

Reply:
That's fair. I think we could restrict integer indices to non-spatial.

In Minc, does a single volume collapse the first dimension, or is would the shape be `(1, x, y, z)`?

Reply:
Could we work out the indexing, and then allow image formats to say whether that kind of indexing is allowed or not, maybe with a call to something like `img._check_slicing()`.  Nifti and Analyze etc could then barf with the integer indices on the spatial axes.